### PR TITLE
DDP-5673 Implement "Thank You" page for Family History

### PR DESCRIPTION
**Pull request by alextanlan**
This change is [Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/480)

### CODE REVIEW

**Review by EdPokhvalin — DISMISSED** [review]
> Reviewed 10 of 15 files at r1.
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/480)* status: 10 of 15 files reviewed, 3 unresolved discussions (waiting on @alextanlan, @fat23cat, and @MocanaAtBroad)

___
*[ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.html, line 1 at r1](https://reviewable.io/reviews/broadinstitute/ddp-angular/480#-MWQ5azk-YPY_eDsCZ8J:-MWQ5azk-YPY_eDsCZ8K:b-2z0etb) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/30bb583b5d4f9ac6fa89ed5a5b2ddf6f0feb38be/ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.html#L1)):*
> ```HTML
> <main class="family-history-thank-you-container">
> ```

This is not a problem, but:
1. According to spec we shouldn't use `<main>` element more than once in a document (although the browser will not give us errors or warnings)
2. Are we considering avoiding additional containers? I mean, the root of the component is already a container
`@Component({ ..., host: {class: 'MyAwsumComponent'}, ... })`. 
___
*[ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.html, line 3 at r1](https://reviewable.io/reviews/broadinstitute/ddp-angular/480#-MWQ3cItCmmVnTF12g3q:-MWQ3cItCmmVnTF12g3r:b-2ozkd3) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/30bb583b5d4f9ac6fa89ed5a5b2ddf6f0feb38be/ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.html#L3)):*
> ```HTML
> <main class="family-history-thank-you-container">
>     <section class="content">
>         First image
> ```

I believe we should translate this alternative text as well
___
*[ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.ts, line 5 at r1](https://reviewable.io/reviews/broadinstitute/ddp-angular/480#-MWQ51T4E7ZZTTlg4n9t:-MWQ51T4E7ZZTTlg4n9u:b3g2fpr) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/30bb583b5d4f9ac6fa89ed5a5b2ddf6f0feb38be/ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.ts#L5)):*
> ```XML
> @Component({
>     selector: 'lib-thank-you-family-history',
>     templateUrl: './thank-you-family-history.component.html',
> ```

Styles seem to be missing here

**Review by alextanlan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/480)* status: 9 of 15 files reviewed, 3 unresolved discussions (waiting on @EdPokhvalin, @fat23cat, and @MocanaAtBroad)

___
*[ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.html, line 1 at r1](https://reviewable.io/reviews/broadinstitute/ddp-angular/480#-MWQ5azk-YPY_eDsCZ8J:-MWTy6Mn3_AFJV1u4Gue:bcqi3en) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/30bb583b5d4f9ac6fa89ed5a5b2ddf6f0feb38be/ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.html#L1)):*
<details><summary><i>Previously, EdPokhvalin wrote…</i></summary><blockquote>

This is not a problem, but:
1. According to spec we shouldn't use `<main>` element more than once in a document (although the browser will not give us errors or warnings)
2. Are we considering avoiding additional containers? I mean, the root of the component is already a container
`@Component({ ..., host: {class: 'MyAwsumComponent'}, ... })`. 
</blockquote></details>

1. I can't see a problem there. `<main>` tag is used once on the ("Thank You Family History") page and contains the main content of the page, imo. 
2. On the one hand, I don't suppose we have a requirement to avoid additional containers. 
On the other hand, styling outside a component a bit more convenient  in this way (if we use classes vs tags as style selector). 
 
___
*[ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.html, line 3 at r1](https://reviewable.io/reviews/broadinstitute/ddp-angular/480#-MWQ3cItCmmVnTF12g3q:-MWTuQ-ADZ2cp-sFd1mJ:b-5dryyw) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/30bb583b5d4f9ac6fa89ed5a5b2ddf6f0feb38be/ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.html#L3)):*
<details><summary><i>Previously, EdPokhvalin wrote…</i></summary><blockquote>

I believe we should translate this alternative text as well
</blockquote></details>

Agree, fixed at 89eaa0f1c65e6bacbbd2bd5ec2fa0abdb4c193d7
___
*[ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.ts, line 5 at r1](https://reviewable.io/reviews/broadinstitute/ddp-angular/480#-MWQ51T4E7ZZTTlg4n9t:-MWTu_S7AtrV0y0neYeE:b-wc8zjx) ([raw file](https://github.com/broadinstitute/ddp-angular/blob/30bb583b5d4f9ac6fa89ed5a5b2ddf6f0feb38be/ddp-workspace/projects/toolkit/src/lib/components/thank-you/thank-you-family-history/thank-you-family-history.component.ts#L5)):*
<details><summary><i>Previously, EdPokhvalin wrote…</i></summary><blockquote>

Styles seem to be missing here
</blockquote></details>

The empty scss file has been removed purposely because all styles for the component are provided in a "parent" activity (for the current case - in ddp-osteo)

File: ddp-workspace/projects/toolkit/src/lib/services/workflowMapper.service.ts [hunk 1]
```diff
@@ -1,31 +1,36 @@
...
     public convert(activityResponse: ActivityResponse): WorkflowAction {
-        if (activityResponse.next === this.toolkitConfiguration.dashboardGuid || activityResponse.next === WorkflowState.DASHBOARD
-            || activityResponse.next === WorkflowState.UNKNOWN) {
+        if (this.DASHBOARD_LIST.includes(activityResponse.next)) {
             return new UrlWorkflowAction(this.toolkitConfiguration.dashboardUrl);
         } else if (activityResponse.next === WorkflowState.PARTICIPANT_LIST) {
             return new UrlWorkflowAction(this.toolkitConfiguration.participantListUrl);
-        } else if (activityResponse.next === this.toolkitConfiguration.lovedOneThankYouGuid
-            || activityResponse.next === WorkflowState.THANK_YOU) {
+        } else if (activityResponse.activityCode === 'FAMILY_HISTORY' && activityResponse.next === WorkflowState.THANK_YOU) {
```
Comment (MocanaAtBroad) [inline ddp-workspace/projects/toolkit/src/lib/services/workflowMapper.service.ts]: I don't think you can count on the response telling you that the activityCode will be FAMILY_HISTORY. You will be told that the state is THANK_YOU and you will have to know that you that user was sent there after completing FAMILY_HISTORY.

Comment (alextanlan) [reply ddp-workspace/projects/toolkit/src/lib/services/workflowMapper.service.ts]: I see that `activityResponse` has these fields
```
 public next: string,
 public activityCode?: string,
 public instanceGuid?: string,
 public allowUnauthenticated?: boolean
```
 Is it possible to get either `activityCode = "FAMILY_HISTORY"` or `next="THANK_YOU_FAMILY_HISTORY"` from back-end in this case ? 
 
Or should I made some sort of a hack (e.g. with localStorage) to keep an info about coming from FAMILY_HISTORY activity?

Comment (alextanlan) [reply ddp-workspace/projects/toolkit/src/lib/services/workflowMapper.service.ts]: I believe I fixed it at 9f3016fbc28ed86e5c76f6fa68eac8da3c4fda12

**Review by MocanaAtBroad — DISMISSED** [review]

**Review by MocanaAtBroad — APPROVED** [review]

**Review by alextanlan — COMMENTED** [review]
> Dismissed @EdPokhvalin and @MocanaAtBroad from 4 discussions.
> *[Reviewable](https://reviewable.io/reviews/broadinstitute/ddp-angular/480)* status: 7 of 18 files reviewed, all discussions resolved (waiting on @EdPokhvalin and @fat23cat)

**Pull request merged**